src/main/kotlin/de/cyface/collector/configuration/Configuration.kt

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2022 Cyface GmbH
+ *
+ * This file is part of the Cyface Data Collector.
+ *
+ * The Cyface Data Collector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.collector.configuration
+
+import io.vertx.core.Future
+import io.vertx.core.Promise
+import io.vertx.core.json.JsonObject
+import io.vertx.kotlin.core.json.get
+import org.apache.commons.lang3.Validate
+import java.net.URL
+import java.nio.file.Path
+import java.nio.file.Paths
+
+/**
+ * A POJO representing the configuration provided to this application.
+ *
+ * @author Klemens Muthmann
+ * @version 1.0.0
+ * @property jwtPrivate The [Path] to the private key file used for JWT authentication.
+ * @property jwtPublic The [Path] to the public key file used for JWT authentication.
+ * @property serviceHttpAddress The [URL] hosting the collector service.
+ * @property mongoDb The Mongo database configuration as described in the Vert.x documentation.
+ * @property adminUser The name of the default admin user to create on startup if it does not exist yet.
+ * @property adminPassword The password for the default user to create if it does not exist. This should be changed in
+ * production.
+ * @property salt The salt to make breaking passwords harder.
+ * @property jwtExpiration The time a JWT token stays valid.
+ * @property uploadExpiration The time an upload session stays valid to be resumed in the future.
```

Comment:
we might want to add the units here:

- jwtExpiration = seconds (int)
- uploadExpiration = milliseconds (long)